sdk/lib/rtl/workitem.c

```diff
@@ -45,899 +45,6 @@ RtlpExitThread(IN NTSTATUS ExitStatus)
-
-                IoThread->Flags |= WT_EXECUTEINPERSISTENTIOTHREAD;
-                PersistentIoThread = IoThread;
-            }
-            else
-            {
-                DPRINT1("Failed to find a worker thread for the persistent IO thread!\n");
-                return STATUS_NO_MEMORY;
-            }
-        }
-    }
-    else
-    {
-        /* Find a worker thread that is not currently executing a long function */
-        CurrentEntry = ThreadPoolIOWorkerThreadsList.Flink;
-        while (CurrentEntry != &ThreadPoolIOWorkerThreadsList)
-        {
-            IoThread = CONTAINING_RECORD(CurrentEntry,
-                                         RTLP_IOWORKERTHREAD,
-                                         ListEntry);
-
-            if (!(IoThread->Flags & WT_EXECUTELONGFUNCTION))
-            {
-                /* if we're trying to queue a long function then make sure we're not dealing
-                   with the persistent thread */
-                if ((WorkItem->Flags & WT_EXECUTELONGFUNCTION) && !(IoThread->Flags & WT_EXECUTEINPERSISTENTIOTHREAD))
-                {
-                    /* found a candidate */
-                    break;
-                }
-            }
-
-            CurrentEntry = CurrentEntry->Flink;
-        }
-
-        if (CurrentEntry == &ThreadPoolIOWorkerThreadsList)
-        {
-            /* Couldn't find an appropriate thread, see if we can use the persistent thread (if it exists) for now */
-            if (ThreadPoolIOWorkerThreads == 0)
-            {
-                DPRINT1("Failed to find a worker thread for the work item 0x%p!\n", WorkItem);
-                ASSERT(IsListEmpty(&ThreadPoolIOWorkerThreadsList));
-                return STATUS_NO_MEMORY;
-            }
-            else
-            {
-                /* pick the first worker thread */
-                CurrentEntry = ThreadPoolIOWorkerThreadsList.Flink;
-                IoThread = CONTAINING_RECORD(CurrentEntry,
-                                             RTLP_IOWORKERTHREAD,
-                                             ListEntry);
-
-                /* Since this might be the persistent worker thread, don't run as a
-                   long function */
-                WorkItem->Flags &= ~WT_EXECUTELONGFUNCTION;
-            }
-        }
-
-        /* Move the picked thread to the end of the list. Since we're always searching
-           from the beginning, this improves distribution of work items */
-        RemoveEntryList(&IoThread->ListEntry);
-        InsertTailList(&ThreadPoolIOWorkerThreadsList,
-                       &IoThread->ListEntry);
-    }
-
-    ASSERT(IoThread != NULL);
-
-    InterlockedIncrement(&ThreadPoolIOWorkerThreadsRequests);
-
-    if (WorkItem->Flags & WT_EXECUTELONGFUNCTION)
-    {
-        /* We're about to queue a long function, mark the thread */
-        IoThread->Flags |= WT_EXECUTELONGFUNCTION;
-
-        InterlockedIncrement(&ThreadPoolIOWorkerThreadsLongRequests);
```

Comment:
Just to know.... Are these metrics present in the new Wine code?

Reply:
Seems so, but that doesn't surprise me much. The code here was from wine originally in the first place.